### PR TITLE
keycloak: 24.0.5 -> 25.0.1

### PR DESCRIPTION
**Pull request by NickCao**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/services/web-apps/keycloak.nix [hunk 1]
```diff
@@ -340,16 +339,13 @@ in
               '';
             };
 
-            hostname-strict-backchannel = mkOption {
```
Comment (teutat3s) [inline nixos/modules/services/web-apps/keycloak.nix:343]: There's another mention of `hostname-strict-backchannel` in `keycloak.md` that'll need updating.

Comment (NickCao) [reply nixos/modules/services/web-apps/keycloak.nix:343]: Done

**Comment by NickCao** [comment]
```
        > caused by:
       >  rendering release-notes/rl-2205.section.md
       >
       > caused by:
       > bad local reference, id #opt-services.keycloak.settings.hostname-strict-backchannel not known
```

What should we do about this?

**Comment by teutat3s** [comment]
Maybe change it to a string `services.keycloak...` without the reference/link to the option?

**Comment by NickCao** [comment]
Marking draft due to some regression: https://github.com/keycloak/keycloak/issues/30306

**Comment by NickCao** [comment]
> Marking draft due to some regression: [keycloak/keycloak#30306](https://github.com/keycloak/keycloak/issues/30306)

This, together with a lot other regressions are now fixed.

**Comment by teutat3s** [comment]
Test nixosTests.keycloak.mysql fails for me locally, not sure if it's the same failure though.

**Review by leona-ya — APPROVED** [review]

File: nixos/modules/services/web-apps/keycloak.nix [hunk 2]
```diff
@@ -340,16 +339,13 @@ in
...
+            hostname-backchannel-dynamic = mkOption {
```
Comment (leona-ya) [inline nixos/modules/services/web-apps/keycloak.nix:342]: I think you might want to add this to the 24.11 release notes.

**Comment by teutat3s** [comment]
Would it be okay to backport this to 24.05? Or would we only do that if security fixes were part of the new version?

**Comment by NickCao** [comment]
> Would it be okay to backport this to 24.05? 

Maybe not.

> EDIT: We did set a precedent with #271845 once.

That was due to the prior version going to EOL right?

**Comment by teutat3s** [comment]
That's [the case](https://endoflife.date/keycloak) here as well, support for 24.0.x ended on 2024-06-10 AFAICT.

**Comment by NickCao** [comment]
> That's [the case](https://endoflife.date/keycloak) here as well, support for 24.0.x ended on 2024-06-10 AFAICT.

Trying out the backport in https://github.com/NixOS/nixpkgs/pull/323480, however the release binary fails to launch for some reason.

**Pull request merged**